docs/xks/developer-guide/best-practices.md

```diff
@@ -3,18 +3,125 @@ id: best-practices
+3. Assume things work and mostly log errors. We have found that most info logs are just a poor man's implementation of metrics and traces. The ideal log is the log that immediately tells you what is wrong with the system. If you can get away with not logging something, consider doing so.
+4. Consider disabling HTTP request logging for non error http requests. We have found that it is common to add a general http log on top of every application. If you have a lot of traffic this will mean a lot of logs. How useful is it really to log that someone made a succesful _GET_ request to a certain endpoint?
+
+### Metrics
+
+Metrics is also a nuanced topic that depends on your context. However, it is usually a good idea to add the so called **RED** metrics to your application. **RED** stand for _Rate_, _Errors_, and _Duration_. We have found that measuring the rate of incoming requests, the rate of them erroring and the time it takes to execute them provides a very good baseline for knowing how your applications are doing.
+
+Consider reading our extended [extended documentation on metrics](https://xenitab.github.io/docs/xks/developer-guide/observability).
+
+### Traces
+
+Add tracing to your application. We have found that modern trace tools provide pretty good configuration out of the box. You just need to add an appropriate tracing library to your application. Consider reading our extended [extended documentation on tracing](https://xenitab.github.io/docs/xks/developer-guide/observability).
+
+## Pod disruption budgets
+
+Make sure you have written a pod disruption budget. This is extensively documented [here](https://xenitab.github.io/docs/xks/developer-guide/scheduling-scaling#pod-disruption-budget).
+A basic one can look as follows
+
+```
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: podinfo
+spec:
+  minAvailable: 2
+  selector:
+    matchLabels:
+      app: podinfo
+```
+
+To check if a budget exists for your application the you can use the following,
+
+```
+kubectl get PodDisruptionBudget
+```
+
+## Network Policies
+
+Configure network policies. No network policy is needed if your application only communicates inside namespace. However, if you want network traffic across namespaces, you need to configure network policies. It is documented extensively [here](https://xenitab.github.io/docs/xks/developer-guide/networking).
+
+## Resources and Scaling
+
+Make sure you have written reasonable resource requests and limits. It is extensively documented [here](https://xenitab.github.io/docs/xks/developer-guide/scheduling-scaling#pod-resources).
+
+## Secret Management
+
+1. Make sure no secrets are commited to either source code repository, nor gitops repository.
+2. Load secrets using Secret Store CSI Driver. It is documented extensively [here](https://xenitab.github.io/docs/xks/developer-guide/secrets-management).
```

Comment:
It's okay to get secrets from Cloud provider key vault solutions as well. As long as you don't save passwords in git you are okay :)